app/auth/authorize_api_request.rb

```diff
@@ -0,0 +1,41 @@
+class AuthorizeApiRequest
+  def initialize(headers = {})
+    @headers = headers
+  end
+
+  # Service entry point - return valid user object
+  def call
+    {
+      user: user
+    }
+  end
+
+  private
+
+  attr_reader :headers
+
+  def user
+    # check if user is in the database
+    # memoize user object
+    @user ||= User.find(decoded_auth_token[:user_id]) if decoded_auth_token
+    # handle user not found
+  rescue ActiveRecord::RecordNotFound => e
+    # raise custom error
+    raise(
+      ExceptionHandler::InvalidToken,
+      ("#{Message.invalid_token} #{e.message}")
+    )
+  end
+
+  # decode authentication token
+  def decoded_auth_token
+    @decoded_auth_token ||= JsonWebToken.decode(http_auth_header)
```

Comment:
Please see the bellow error which occurs on run time and fails the project to run.
![jwt](https://user-images.githubusercontent.com/35504149/146959634-589a929b-4a61-4f95-a710-6c17322a3d0d.png)